server/src/test/java/org/elasticsearch/common/geo/GeometryIndexerTests.java

```diff
@@ -559,6 +561,30 @@ public void testIssue82840() {
         assertEquals(indexedCW, indexerCW.prepareForIndexing(polygon));
     }
 
+    public void testPolygonLuceneTessellationBug() throws IOException, ParseException {
+        GeoShapeIndexer indexer = new GeoShapeIndexer(true, "test");
+        String wkt = "POLYGON((-88.3245325358123 41.9306419084828,-88.3243288475156 41.9308130944597,-88.3244513948451 41.930891654082,"
+            + "-88.3246174067624 41.930998076295,-88.3245448815692 41.9310557712027,-88.3239353718069 41.9313272600886,"
+            + "-88.3237355617867 41.9313362704162,-88.3237347670323 41.9311150951881,-88.3237340649402 41.931103661118,"
+            + "-88.3235660813522 41.9311112432041,-88.3234509652339 41.9311164377155,-88.3232353124097 41.9311261692953,"
+            + "-88.3232343331295 41.9313588701899,-88.323028772523 41.9313681383084,-88.3229999744274 41.930651995613,"
+            + "-88.3236147717043 41.9303655647412,-88.323780013667 41.929458561339,-88.3240657895016 41.9293998882959,"
+            + "-88.3243948640426 41.9293028003164,-88.324740490767 41.9301340399879,-88.3251305560187 41.9302766363048,"
+            + "-88.3248260581475 41.9308286995884,-88.3246595186817 41.9307227160738,-88.3245325358123 41.9306419084828),"
+            + "(-88.3245658060855 41.930351580587,-88.3246004191532 41.9302095159456,-88.3246375011905 41.9300573183932,"
+            + "-88.3243392233337 41.9300159738164,-88.3243011787553 41.9301696594472,-88.3242661951392 41.9303109843373,"
+            + "-88.3245658060855 41.930351580587),(-88.3245325358123 41.9306419084828,-88.3245478066552 41.9305086556331,"
+            + "-88.3245658060855 41.930351580587,-88.3242368660096 41.9303327977821,-88.3242200926128 41.9304905242189,"
+            + "-88.324206161464 41.9306215207536,-88.3245325358123 41.9306419084828),(-88.3236767661893 41.9307089429871,"
+            + "-88.3237008716322 41.930748885445,-88.323876104365 41.9306891087739,-88.324063438129 41.9306252050871,"
+            + "-88.3239244290607 41.930399373909,-88.3237349076233 41.9304653056436,-88.3235653339759 41.9305242981369,"
+            + "-88.3236767661893 41.9307089429871))";
+        Geometry geometry = WellKnownText.fromWKT(GeographyValidator.instance(true), false, wkt);
+        // With the lucene bug this next line will throw an IllegalArgumentException
+        List<IndexableField> fields = indexer.indexShape(geometry);
```

Comment:
We have that error in the tessellator tests. Do you need that kind of test in both places?

What I have in the tessellator are negative tests for all three bugs, since we also have the positive tests for all three there too.

Reply:
ah ok, so let's remove that comment as it makes very little sense as it says a lucene bug without any context?

Reply:
Fixed in 6f7518e56ae2280ebed6d89b34f16f64342f3875